src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -549,23 +546,24 @@ private async Task ConnectFirstAsync(CancellationToken cancellationToken)
                 }
                 catch (Exception e) // gather exceptions
                 {
-                    exceptions ??= new List<Exception>();
-                    exceptions.Add(e);
+                    _logger.LogError(e, "Connection attempt has thrown.");
                     canRetry = false; // retry strategy threw
+                    isExceptionThrown = true;
                 }
 
             } while (canRetry);
 
-            var aggregate = exceptions == null ? null : new AggregateException(exceptions);
+            string msgSomeThingWentWrong = "";
+
+            if (isExceptionThrown)
+                msgSomeThingWentWrong = "Exception(s) is thrown in the past. Please, see the logs.";
 
             // canceled exception?
             if (cancellationToken.IsCancellationRequested)
-                throw new OperationCanceledException($"The cluster connection operation to \"{_clusterState.ClusterName}\" has been canceled. " +
-                    $"The following addresses where tried: {string.Join(", ", tried)}.", aggregate);
```

Comment:
I think we want to keep logging the addresses that were tried. Can you keep the "The following addresses..." message? And verify that the `tried` thing is some sort of `HashSet<>` that will not grow infinitely?

Reply:
I undo that part. Tried addresses are already kept in set.